src/tactic/interactive_expr.lean

```diff
@@ -55,15 +59,15 @@ meta instance : has_repr sf := ⟨λ s, s.repr.to_string⟩
 /-- Constructs an `sf` from an `eformat` by forgetting grouping, nesting, etc. -/
 meta def sf.of_eformat : eformat → sf
 | (tag ⟨ea,e⟩ m) := sf.tag_expr ea e $ sf.of_eformat m
-| (group m) := sf.of_eformat m
-| (nest i m) := sf.of_eformat m
-| (highlight i m) := sf.of_eformat m
+| (group m) := sf.block 0 $ sf.of_eformat m
```

Comment:
@gebner do you know if this is the right semantics for `group`? Eg in the pp'er for `list`, the inner format is wrapped in both a group and a list. So this might create excess spans. Can we get away with just mapping `group m` to `sf.of_eformat m`?

Reply:
The semantics of `group` is:
 1. If the content fits on the current line, print all the `format.line` inside the group as spaces.
 2. If it doesn't fit on the current line, print `format.line` as newlines.

`nest` then specifies how many spaces should be inserted after the newline.

Reply:
From what I understood, `group` means the same a `nest`, but without indentation. What it means, is that a newline may or may not happen, but it tells that those belong together.

Reply:
The demo looked ok, or at least better than what we have right now.  I'm not sure if it's possible to implement a Wadler-Leijen pretty printer in pure CSS.

Reply:
`nest` and `group` are independent, although they are typically used together.  For example if you only use `nest`, then nothing happens.